src/core/search/searchdatatransformer.js

```diff
@@ -31,16 +32,44 @@ export default class SearchDataTransformer {
   }
 
   static transformVertical (data, formatters, verticalKey) {
-    // TODO (tmeyer): SPR-1923, centralize the allResultsForVertical parsing here.
+    const resultsContext = SearchDataTransformer._determineResultsContext(data.response);
+    const response = SearchDataTransformer._formResponseWithContext(data.response, resultsContext);
```

Comment:
This `_formResponseWithContext` only works with vertical responses, do you think it should have `vertical` in the function signature?

Reply:
sure sgtm!